nixops/backends/libvirtd.py

```diff
@@ -181,22 +202,19 @@ def iface(n):
         )
 
     def _parse_ip(self):
-        cmd = [
-            "virsh",
-            "-c",
-            "qemu:///system",
-            "net-dhcp-leases",
-            "--network",
-            self.primary_net,
-        ]
-        lines = subprocess.check_output(cmd)
-        try:
-            i = lines.split().index(self.primary_mac)
-        except ValueError:
-            pass
-        else:
-            ip_with_subnet = lines.split()[i + 2]
-            return ip_with_subnet.split('/')[0]
+        """
+        return ip
+        """
+        # alternative is VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE
+        ifaces = self.dom.interfaceAddresses(libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE, 0)
+        if (ifaces == None):
+            self.log("Failed to get domain interfaces")
+            sys.exit(0)
```

Comment:
Shouldn't this be exit(1) or something? Also, the current code doesn't call exit() at all. I think adding a comment (in the commit message) about why adding a process exit point in this function is a good thing. (I.e. is it a bug in the current code that it does _not_ call exit?)

Reply:
that's an oversight from my side, I wanted to put it back to "None" as in the original code and forgot it.